src/util/identifier.js

```diff
@@ -31,7 +31,7 @@ export const reservedWords = {
 // And the keywords
 
 export const isKeyword = makePredicate(
-  "break case catch continue debugger default do else finally for function if return switch throw try var while with null true false instanceof typeof void delete new in this let const class extends export import yield super",
+  "break case catch continue debugger default do else finally for function if return switch throw try var while with null true false instanceof typeof void delete new in this let const class extends export import yield super match",
```

Comment:
@hzoo @xtuc I wonder if we should make `match` a keyword at all (at least part of this array)? 
If we do it then we need to tweak for all situations where match is currently being used as a valid identifier.

What I imagined instead as a simplest implementation, is to tokenize it as a identifier and use lookup to determine if it should be a keyword in a given context (part of match expression or just a ex. function call `match(something);`).